model/src/main/kotlin/utils/FileProvenanceFileStorage.kt

```diff
@@ -81,6 +82,7 @@ private fun KnownProvenance.hash(): String {
     val key = when (this) {
         is ArtifactProvenance -> "${sourceArtifact.url}${sourceArtifact.hash.value}"
         is RepositoryProvenance -> "${vcsInfo.type}${vcsInfo.url}$resolvedRevision"
+        is DirectoryProvenance -> ""
```

Comment:
I don't think it's a good idea to (even temporarily) return an empty string here. We could consider to use an SPDX package verification code on the directory here. However, that's potentially expensive to compute. Why not just use the `canonicalPath` property? That's at least better than the empty string.